src/shared/identityUtils.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+
+import { CryptoUtils } from '@salesforce/lwc-dev-mobile-core';
+import { LightningDevConfig } from './lightningDevConfig.js';
+import { ConfigVars } from './../configMeta.js';
+
+export class IdentityUtils {
+  public static async updateConfigWithIdentityToken(): Promise<void> {
+    const token = await this.getIdentityToken();
+    if (!token) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+      const generatedIdentityToken = CryptoUtils.generateIdentityToken();
+      await this.writeIdentityToken(generatedIdentityToken);
+    }
+    return Promise.resolve();
+  }
+
+  public static async getIdentityToken(): Promise<string | undefined> {
+    const config = await LightningDevConfig.create({
+      isGlobal: false,
+    });
+    await config.read();
+    const identityToken = config.get(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_TOKEN);
+
+    if (identityToken) {
+      const identityTokenAsString = identityToken as string;
+      return Promise.resolve(identityTokenAsString);
+    }
+
+    return Promise.resolve(undefined);
+  }
+
+  public static async writeIdentityToken(token: string): Promise<void> {
```

Comment:
This method could be `private` but when I do that I won't be able to stub it in the unit test.

Reply:
I think it is ok to leave it `public`.... it is a function in a utility library that could be on use in the future

Reply:
I would however update the method signature to take an optional `config` param in addition to the token.

Reply:
@maliroteh-sf I didn't add optional `config` in "write" because now we need to deal separate "configs" depending on "writing" to or "reading" from `sf`'s configuration. Making `config` optional and delegate that responsibility to the called is certainly an approach that could have been taken. But with given inflexibility with `Config`'s APIs and their shortcomings I think the safest way for now for us is to encapsulate the different  configs(`Config` and `ConfigAggregator`) in respective "write" and "read" methods.